client/components/modal/modal.service.js

```diff
@@ -207,6 +207,32 @@ app.factory('Modal', function($rootScope, $uibModal) {
         },
         model: {}
       }, 'modal-success', 'components/modal/form-modal.html');
+    },
+
+    confirmDelete: function(title, templateUrl, model, cb) {
+      var confirmDelete = openModal({
+        modal: {
+          dismissable: true,
+          title: title,
+          templateUrl: templateUrl,
+          submitFn: function(form, model) {
+            cb(model)
+            confirmDelete.close();
+          },
```

Comment:
Hmm actually this could require the cb to return a promise. Right now it just closes the modal, if we want to block the ui until the server returns, we could handle the cb the same way the `form` model does and closes the modal when the `cb` promise resolves.